ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/util/AuthenticationUtil.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2022 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.rest.util;
+
+import io.confluent.ksql.security.KsqlAuthTokenProvider;
+import io.confluent.ksql.security.KsqlPrincipal;
+import io.confluent.ksql.util.KsqlConfig;
+import java.time.Clock;
+import java.util.Objects;
+import java.util.Optional;
+
+public class AuthenticationUtil {
+
+  final Clock clock;
+
+  public AuthenticationUtil(final Clock clock) {
+    this.clock = Objects.requireNonNull(clock);
+  }
+
+  public Optional<Long> getTokenTimeout(
+      final Optional<KsqlPrincipal> principal,
+      final KsqlConfig ksqlConfig,
+      final Optional<KsqlAuthTokenProvider> authTokenProvider
+  ) {
+    final long maxTimeout =
+        ksqlConfig.getLong(KsqlConfig.KSQL_WEBSOCKET_CONNECTION_MAX_TIMEOUT_MS);
+    if (maxTimeout > 0
+        && authTokenProvider.isPresent()
+        && principal.isPresent()
+        && authTokenProvider.get().getLifetimeMs(principal.get()).isPresent()
+    ) {
+      final long tokenTimeout = authTokenProvider.get().getLifetimeMs(principal.get()).get()
+          - clock.millis();
+      return Optional.of(Math.min(tokenTimeout, maxTimeout));
+    } else {
+      return Optional.empty();
+    }
```

Comment:
If `KSQL_WEBSOCKET_CONNECTION_MAX_TIMEOUT_MS` is a positive number, but there's not an Auth provider or other principal details, shouldn't the connection timeout be set to the maximum used in the config instead of empty? Because we're not linking this config with Auth, then I think we should use the MAX config in that case.

Reply:
Yeah that makes sense